README.md

```diff
@@ -1 +1,86 @@
-# Arcade Shared Interfaces
+# Arcade Project - Shared Interfaces
+
+## Introduction
+This document outlines the shared part for the Arcade project, that will be used by both Group. The purpose of these interfaces is to ensure compatibility between the two groups components, and for loading shared libraries using DLLoaders.
+
+## Extern "C" Structure
+To facilitate compatibility between the two groups shared libraries, both groups must adhere to the following structure when exporting symbols:
+
+```cpp
+extern "C" {
+    // Mandatory symbols for drivers
+    std::unique_ptr<IDriver> create_driver(void);
+
+    // Mandatory symbols for games
+    std::unique_ptr<IGame> create_game(void);
+
+    // Optional symbol
+    const std::string &get_name(void);
+
+    // Optionals symbols for loading and unloading the shared library
+    __attribute__((constructor))
+    void load_lib(void);
+    __attribute__((destructor))
+    void unload_lib(void);
+}
+```
+
+### Symbols Description
+- `create_driver()`: This function is mandatory for drivers. It should return a unique pointer to the driver instance.
+- `create_game()`: This function is mandatory for games. It should return a unique pointer to the game instance.
+- `get_name()`: This function is optional. It should return a constant reference to a string containing the name of the component.
+- `load_lib()`: This function is optional. It is called when the shared library is loaded.
```

Comment:
I think adding some text here to explain that the game should not be treated as started or the window has opened, would be a good idea.

```suggestion
- `load_lib()`: This function is optional. It is called when the shared library is loaded. (For a game or a driver this doesn't automatically imply that the game/driver has been launched. Just that the .so has been opened)
```